enhancements/sdk-supports-operands-deletion.md

```diff
@@ -0,0 +1,118 @@
+
+## Motivation
+
+An operand is any resource that an operator manages/reconciles/operates on.
+For this cleanup proposal’s perspective the Custom Resource (CR) object managed by the operator is the only resource object that is considered for cleanup. Other resources such as pods, volumes and off-cluster resources created by the operator are also operands but those resources will get auto-cleaned as a result of removing the CRs.
+
+In kubernetes garbage collection model, some objects are owners of other objects based on the resource hierarchy. Dependent objects have [OwnerReference][owner-reference] field that points to the owning object. This ownership can also be specified manually by the user. If user deletes the owner object, Kubernetes garbage collector will delete all of its dependent objects (sometimes automatically if it is specified).
+
+However, cross-namespaced owner references are not allowed by design and this is where the operator would need to leverage using [Finalizers][finalizers]. to perform cleanup of operands by processing certain deletion events.
+
+SDK will encourage and enable the use of finalizers when it is required to. By default, all operators would have scaffolded logic for Finalizers.
+
+## Goals
+
+- Provide default scaffolding for an operand cleanup
+- Cleanup is strictly on an opt-in basis only and not automatic by default
+- Cleanup is limited to removal of CRs and not CRDs
+- SDK will have additional flags to prevent deletion of CRDs and instead use OLM's operand cleanup feature
+
+## Non-Goals
+
+- SDK support in handling cleaning up resources other than CRs  
+
+## Possible Future Goals (Not in scope)
+
+- SDK should have a mechanism to handle events/alerts received by OLM in case cleanup is blocked
+- The timeout for operand deletion should be user configurable by the SDK
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator author, I want an upstream finalizer no-op handler scaffolded by default and/or library to help write finalizers.
+
+#### Story 2
+
+As an operator author, I want scaffolded fields in the CSV generation by default to enable operand cleanup.
+
+#### Story 3
+
+As an operator author, I would like the Operator SDK's cleanup command to handle deletion of CRDs appropriately based on the flags provided.
+
+### Implementation Details/Notes/Constraints
+
+#### Upstream finalizer no-op handler scaffolded by default and/or library to help write Finalizers
+
+Any operator that handles delete events or if the operator is expecting a delete event for CRs should have Finalizers. The only time Finalizers may not be required is when OwnerReferences exist and they handle all of the cleanup. But as we tend towards cluster scoped operators, OwnerReferences won't be allowed. In that case, all operators should have Finalizers.
```

Comment:
```suggestion
Any operator that deletes or expects delete events for CRs that create resources can use finalizers or owner references, although the latter can be used when those CRs only create resources in the same namespace. Additionally finalizers permit customizable deletion steps, ex. update a CR's status during deletion with a particular message. Because finalizers are broadly applicable and give operator authors control over how CR deletion proceeds, operator projects would benefit from some initial finalizer code scaffolded by default.
```